ethportal-api/src/types/state_trie/utils.rs

```diff
@@ -0,0 +1,75 @@
+use eth_trie::node::Node;
+use keccak_hash::{keccak, H256};
+use rlp::RlpStream;
+
+/// Our own implementation for encoding the node.
+///
+/// It is almost identical copy from the `eth_trie` crate, because that one is not reusable.
```

Comment:
Not really. There are no tests that test this logic directly (instead they test entire Trie structure and this is tested as part of it).
I would rather not spend time on writing tests for this because:
- this logic is also tested indirectly by state validator logic
- I will want to rewrite library for the trie validation and not use `eth_trie` at all